idletiming_conn.go

```diff
@@ -53,6 +69,7 @@ func Conn(conn net.Conn, idleTimeout time.Duration, onIdle func()) *IdleTimingCo
 				continue
 			case <-timer.C:
 				c.Close()
+				atomic.StoreInt64(&c.idled, 1)
```

Comment:
`c.Close()` sets `c.closed` to true and closes the wrapped connection. All r/w afterward returns `ErrIdled` (except the first `Read()` which returns `EOF`?), which is already a good indicator. In which case we need to actively check `IsIdled()`?

Reply:
We've been seeing a lot of EOF errors in our probes and I'm curious if any of them are caused by a connection having been idled. That's what I need IsIdled() for.

Reply:
I see. LGTM. Merging